### PR TITLE
UCX + Dask Doc

### PR DESCRIPTION
**Pull request by quasiben**
This PR adds documentation on using dask with UCX

cc @jakirkham @pentschev

### CODE REVIEW

**Comment by pentschev** [comment]
rerun tests

**Comment by quasiben** [comment]
rerun tests

File: docs/source/dask.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    )
+
+    cluster = DGX(
+        interface="enp1s0f0",
+        protocol="ucx",
+        enable_tcp_over_ucx=enable_tcp_over_ucx,
+        enable_infiniband=enable_infiniband,
+        enable_nvlink=enable_nvlink,
+    )
+    client = Cluster(client)
```
Comment (pentschev) [inline docs/source/dask.rst]: Maybe add a small note to explain that the interface must be the ethernet interface.

File: docs/source/dask.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    # worker
+    UCX_CUDA_IPC_CACHE=n UCX_TLS=tcp,sockcm,cuda_copy,cuda_ipc \
+    UCX_SOCKADDR_TLS_PRIORITY=sockcm dask-cuda-worker ucx://{SCHEDULER_ADDR}:8786
+
+    # client
+    UCX_CUDA_IPC_CACHE=n UCX_TLS=tcp,cuda_copy,cuda_ipc,sockcm \
+    UCX_SOCKADDR_TLS_PRIORITY=sockcm python <python file>
+
+
+The benefit of using ``dask-cuda-worker`` is that it will invoke N workers where N is the number of GPUs and automatically pair workers with GPUs.
```
Comment (pentschev) [inline docs/source/dask.rst]: Also here, same as above, small note regarding ethernet interface.

File: docs/source/dask.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    # worker
+    CUDA_VISIBLE_DEVICES=0 UCX_CUDA_IPC_CACHE=n UCX_TLS=tcp,sockcm,cuda_copy,cuda_ipc \
+    UCX_SOCKADDR_TLS_PRIORITY=sockcm dask-worker ucx://{SCHEDULER_ADDR}:8786
+
+    # client
+    UCX_CUDA_IPC_CACHE=n UCX_TLS=tcp,cuda_copy,cuda_ipc,sockcm \
+    UCX_SOCKADDR_TLS_PRIORITY=sockcm python <python file>
+
+Note: ``CUDA_VISIBLE_DEVICES`` controls which GPU(s) the worker has access to.
```
Comment (pentschev) [inline docs/source/dask.rst]: I would say add another one here, but maybe we should have a single note about the ethernet interface and explain that it should be used for the cases of all examples.

**Comment by pentschev** [comment]
Added a few small comments, but looks good overall. Thanks for adding those @quasiben !

**Comment by quasiben** [comment]
@pentschev thanks for the review!  I think I've all addressed your comments

File: docs/source/dask.rst [hunk 4]
```diff
@@ -72,6 +75,7 @@ Lastly, we can also manually start each worker individually (this is typically o
 .. code-block:: bash
 
     # server
+    # Note:
```
Comment (pentschev) [inline docs/source/dask.rst]: Is this line supposed to be here or just a leftover?

Comment (quasiben) [reply docs/source/dask.rst]: No.  Remove!

**Comment by pentschev** [comment]
Cool, thanks for addressing those @quasiben . :)

**Comment by quasiben** [comment]
rerun tests

**Comment by quasiben** [comment]
rerun tests

**Pull request merged**